### ISSUE TITLE
Replica dying when data is in flight causes writes to fail.

### ISSUE THREAD

**Issue by sangeethah** [rancher/longhorn#85]
rancher-server version - v1.1.0-dev2-rc3
longhorn version - rancher/longhorn:v0.1.0

Steps to reproduce the problem:
Deploy VM with ROOT and DATA disk on longhorn.

Use add command to write data into the DATA disk (dd if=/dev/zero of=test bs=128)
Have this command running for 30-45 seconds.
When this command is still executing , power off the host where one of the replica is running.

dd command quits with the following error:
<img width="970" alt="screen shot 2016-05-15 at 11 48 17 am" src="https://cloud.githubusercontent.com/assets/4266958/15299056/6f76d1d4-1b57-11e6-9b10-4a28b4e345fa.png">

After this error , I am still able read/write to data disk. 

We should see write errors when one of the replica goes down.

**Comment by sangeethah** [comment]
rancher-server version - Build from master
longhorn version -  cjellick/longhorn:dev-147d83f

Steps to reproduce the problem:
Deploy VM with ROOT and DATA disk on longhorn.

Use dd command to write data into the DATA disk (dd if=/dev/zero of=test bs=128)
Have this command running for 30-45 seconds.

Killed one of the replica process.
dd command itself did not quit.

But when I tried to do a sync , it hung for a long time and after a while I see the following error on console:

<img width="903" alt="screen shot 2016-05-20 at 1 27 30 pm" src="https://cloud.githubusercontent.com/assets/4266958/15441250/b6f2f9f4-1e8e-11e6-967a-c9da5347c3eb.png">

Controller itself reports both replicas in correct state:
<img width="661" alt="screen shot 2016-05-20 at 1 31 48 pm" src="https://cloud.githubusercontent.com/assets/4266958/15441350/3b633672-1e8f-11e6-89a5-6c281a711ae7.png">


```
mysql> select id,name,state,health_state,created,removed from instance where name like "%vm1%";
+------+--------------------------------------------------------------------+---------+----------------+---------------------+---------------------+
| id   | name                                                               | state   | health_state   | created             | removed             |
+------+--------------------------------------------------------------------+---------+----------------+---------------------+---------------------+
| 2549 | Default_vm1_1                                                      | running | NULL           | 2016-05-20 19:04:05 | NULL                |
| 2550 | volume-Default-vm1-1-325be6f-root_controller_1                     | running | healthy        | 2016-05-20 19:04:15 | NULL                |
| 2551 | volume-Default-vm1-1-325be6f-root_replica_replica-binary_1         | stopped | NULL           | 2016-05-20 19:04:15 | NULL                |
| 2552 | volume-Default-vm1-1-325be6f-root_controller_controller-agent_1    | running | NULL           | 2016-05-20 19:04:19 | NULL                |
| 2553 | volume-Default-vm1-1-325be6f-root_replica_1                        | running | healthy        | 2016-05-20 19:04:39 | NULL                |
| 2554 | volume-Default-vm1-1-325be6f-root_replica_replica-healthcheck_1    | running | NULL           | 2016-05-20 19:04:42 | NULL                |
| 2555 | volume-Default-vm1-1-325be6f-root_replica_sync-agent_1             | running | NULL           | 2016-05-20 19:04:42 | NULL                |
| 2556 | volume-Default-vm1-1-325be6f-root_replica_replica-binary_2         | stopped | NULL           | 2016-05-20 19:04:43 | NULL                |
| 2557 | volume-Default-vm1-1-325be6f-root_replica_2                        | running | healthy        | 2016-05-20 19:05:11 | NULL                |
| 2558 | volume-Default-vm1-1-325be6f-root_replica_replica-healthcheck_2    | running | NULL           | 2016-05-20 19:05:26 | NULL                |
| 2559 | volume-Default-vm1-1-325be6f-root_replica_sync-agent_2             | running | NULL           | 2016-05-20 19:05:26 | NULL                |
| 2560 | volume-Default-vm1-1-325be6f-testme1_controller_1                  | running | healthy        | 2016-05-20 19:05:58 | NULL                |
| 2561 | volume-Default-vm1-1-325be6f-testme1_replica_1                     | running | healthy        | 2016-05-20 19:05:58 | NULL                |
| 2562 | volume-Default-vm1-1-325be6f-testme1_replica_replica-healthcheck_1 | running | NULL           | 2016-05-20 19:06:02 | NULL                |
| 2563 | volume-Default-vm1-1-325be6f-testme1_controller_controller-agent_1 | running | NULL           | 2016-05-20 19:06:02 | NULL                |
| 2564 | volume-Default-vm1-1-325be6f-testme1_replica_sync-agent_1          | running | NULL           | 2016-05-20 19:06:02 | NULL                |
| 2565 | volume-Default-vm1-1-325be6f-testme1_replica_2                     | purged  | reinitializing | 2016-05-20 19:06:04 | 2016-05-20 20:14:17 |
| 2566 | volume-Default-vm1-1-325be6f-testme1_replica_replica-healthcheck_2 | purged  | NULL           | 2016-05-20 19:06:07 | 2016-05-20 20:14:18 |
| 2567 | volume-Default-vm1-1-325be6f-testme1_replica_sync-agent_2          | purged  | NULL           | 2016-05-20 19:06:08 | 2016-05-20 20:14:18 |
| 2568 | volume-Default-vm1-1-325be6f-testme1_replica_2                     | running | healthy        | 2016-05-20 20:14:17 | NULL                |
| 2569 | volume-Default-vm1-1-325be6f-testme1_replica_replica-healthcheck_2 | running | NULL           | 2016-05-20 20:14:21 | NULL                |
| 2570 | volume-Default-vm1-1-325be6f-testme1_replica_sync-agent_2          | running | NULL           | 2016-05-20 20:14:21 | NULL                |
+------+--------------------------------------------------------------------+---------+----------------+---------------------+---------------------+
22 rows in set (0.00 sec)
```

**Comment by kp6** [comment]
@sangeethah I spent a fair amount of time trying to reproduce this without rancher but controller serves dd writes just fine so far: No filesystem related errors at all.
So I wonder what's the best path to proceed further? 
1. I could try more simulations of rancher behavior by scripting or manually performing replica remove, add/sync for the faulty replica
2. You or I could try reproducing this on your setup and monitor the controller logs 
3. I make my own rancher/packet setup work, but it may take a while as this would be my first time to setup everything

would option 2 above work for you?
- in other words we need the controller logs to move this further. The fastest way to get them would be either:
 - If you let me to borrow your setup and show me how to reproduce this and get to the controller container logs
 - or you reproduce it and give me the logs

please let me know what works for you

**Comment by sangeethah** [comment]
@kp6 , I can provide you a rancher host setup with packets 4 hosts for you to try out this case tomorrow morning.
This issue does not reproduce always for me. It happens once in a while.

**Comment by kp6** [comment]
@sangeethah 
Unreproducible. 
Not sure what's our procedure for issues like this?
